server/src/middlewares/blogs.js

```diff
@@ -1,31 +1,105 @@
-// eslint-disable-next-line arrow-body-style
-const blogExists = async (req, res, next) => {
-  // TODO: Validate whether blog exists or not
-
-  /**
-        * Validation steps.
-        * 1. Check whether the blog exist or not using the blog service.
-        * 2. If the blog does not exists send response as 404: Not Found.
-        * 3. If the blog is found forward the request to the next middleware.
-        */
+import mongoose from 'mongoose';
+import Blogs from '../models/blogs';
+import { blogCreation as blogCreationValidation } from '../utils/validations';
+
+const authenticateUserBlog = async (req, res, next) => {
+  const userID = req.user._id;
+  const blogOwnerId = req.blog.userId;
+  if (userID.toString() !== blogOwnerId.toString()) {
+    return res.status(401).json({ msg: 'Unauthorized Access' });
+  }
+
   return next();
 };
 
-// eslint-disable-next-line arrow-body-style
-const authenticateUserBlog = async (req, res, next) => {
-  // TODO: Validate whether or not user owns the blog or not
-
-  /**
-    * Validation steps.
-    * 1. Get the user id from the request object.
-    * 2. Use blog service to check whether the user owns the blog or not.
-    * 3. If the blog does not belong to user, return status as 401: Unauthorized Access.
-    * 4. Otherwise forward the request to next middleware.
-    */
+const createBlogRequestValidation = async (req, res, next) => {
+  const blog = req.body;
+  const userFieldsValidity = blogCreationValidation(blog);
+  if (!userFieldsValidity.isValid) {
+    res.status(400);
+    return res.json(userFieldsValidity.errors);
+  }
   return next();
 };
 
+const saveBlog = (req, res, next) => {
+  const { title } = req.body;
+  const { content } = req.body;
+  const userId = req.user._id;
+
+  const blog = {
+    title,
+    content,
+    userId,
+  };
+
+  Blogs.create(blog, (err, blogFromDB) => {
+    if (err) {
+      res.status(500).json({ msg: 'Something went wrong', error: err });
+    }
+    req.blog = blogFromDB;
+    next();
+  });
+};
+
+const getBlog = (req, res, next) => {
+  const blogID = req.params.id;
+  const validObjectId = mongoose.Types.ObjectId.isValid(blogID);
+
+  if (validObjectId) {
+    Blogs.findById(blogID, (err, blog) => {
+      if (err) {
+        return res.status(500).json({ msg: 'Something went wrong', error: err });
+      }
+      if (blog) {
+        req.blog = blog.toObject();
+        next();
+      } else {
+        return res.status(404).json({ msg: "can't find blog with the given id." });
+      }
+    });
+  } else {
+    return res.status(400).json({ msg: "Can't find post, invalid url" });
+  }
+};
+
+const updateBlog = (req, res, next) => {
+  const blog = { ...req.blog };
+  blog.title = req.body.title;
+  blog.content = req.body.content;
+  blog.private = req.body.private;
+
+  Blogs.findByIdAndUpdate(req.params.id, blog, { new: true }, (err, blogFromDB) => {
+    if (err) {
```

Comment:
To deal with any database calls I would suggest using service. It would reduce the dependency of mongoose models on the middleware.

Reply:
How will it reduce dependency? The logic will be moved to a file called service.

Reply:
Think of dealing with queries in every file of your project. Now if in future someone wants to remove the dependency of mongoose so he/she would have to go through each file and without knowing where we are dealing with the query logic. So in that, the person needs to look into each and every file but if we do the query part in one type of file only it would become easy to replace and there will be no guessing where the queries are.